src/python/WMComponent/WorkflowUpdater/WorkflowUpdaterPoller.py

```diff
@@ -463,7 +464,10 @@ def findWflowsWithPileup(listSpecs):
                 workloadHelper.load(wflowSpec['spec'])
                 pileupSpecs = workloadHelper.listPileupDatasets()
                 if pileupSpecs:
-                    wflowSpec['pileup'] = pileupSpecs.values()
+                    puSet = set()
```

Comment:
Todor, let me suggest a more readable variation to this solution:
```
wflowSpec['pileup'] = set()
for pileupN in pileupSpecs.values():
    wflowSpec['pileup'].add(pileupN)
```

Variables with less than 3 chars are usually unwanted as well.

Reply:
@amaltaro  This suggestion is not going to work. Here is the error it fails with:
```
2024-04-26 20:01:28,199:139921118983936:ERROR:WorkflowUpdaterPoller:Failed to load spec file for: /data/srv/wmagent/v2.3.2rc8/install/wmagentpy3/WorkQueueManager/cache/tivanov_SC_MultiPU_Feb2024_Val_PartPU_v7_240419_220203_5072/WMSandbox/WMWorkload.pkl. Details: unhashable type: 'set'
```

And the explanation is simple: you cannot add  a mutable  object (a view) as a member of a set. All members of a set needs to be `hashable`, while the members of a `view` are not. I've been facing this very same error in few variants of this patch, until I realized our only way out is to iterate through the already present sets inside the view and make a union with the zero set. 

checking it manually gives the same error:

```
In [138]: print(pileupSpecs)
{'https://cmsweb-testbed.cern.ch/dbs/int/global/DBSReader': {'/Neutrino_E-10_gun/RunIISummer20ULPrePremix-UL16_106X_mcRun2_asymptotic_v13-v1/PREMIX', '/RelValMinBias_14TeV/CMSSW_11_2_0_pre8-112X_mcRun3_2024_realistic_v10_forTrk-v1/GEN-SIM'}}

In [139]: wflowSpec = {}

In [140]: wflowSpec['pileup'] = set()

In [141]: for pileupN in pileupSpecs.values():
     ...:     wflowSpec['pileup'].add(pileupN)
     ...: 
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[141], line 2
      1 for pileupN in pileupSpecs.values():
----> 2     wflowSpec['pileup'].add(pileupN)

TypeError: unhashable type: 'set'

```

I can change the value names, though, such that they are not two chars long.

Reply:
this  on the other hand ... works: 

```
In [143]: for pileupN in pileupSpecs.values():
     ...:     wflowSpec['pileup'] =  wflowSpec['pileup'] | pileupN
     ...: 

In [144]: wflowSpec['pileup']
Out[144]: 
{'/Neutrino_E-10_gun/RunIISummer20ULPrePremix-UL16_106X_mcRun2_asymptotic_v13-v1/PREMIX',
 '/RelValMinBias_14TeV/CMSSW_11_2_0_pre8-112X_mcRun3_2024_realistic_v10_forTrk-v1/GEN-SIM'}

```


I am going to update the patch. I like it better, indeed.

Reply:
hi @amaltaro  - Done. And tested. Please take another look.